build.rs

```diff
@@ -58,83 +58,83 @@ fn ossl_bindings(header: &str, args: &[&str], out_file: &Path) {
 
 #[cfg(not(feature = "dynamic"))]
 fn build_ossl(out_file: &Path) {
-    let openssl_path = std::path::PathBuf::from("openssl")
-        .canonicalize()
-        .expect("cannot canonicalize path");
+    let openssl_path =
+        std::path::PathBuf::from(env!("KRYOPTIC_OPENSSL_SOURCES"))
+            .canonicalize()
+            .expect("cannot canonicalize OpenSSL path");
+
+    let mut buildargs = vec![
+        "no-deprecated",
+        "no-aria",
+        "no-argon2",
+        "no-atexit",
+        "no-des",
+        "no-dsa",
+        "no-cast",
+        "no-mdc2",
+        "no-ec2m",
+        "no-rc2",
+        "no-rc4",
+        "no-rc5",
+        "no-rmd160",
+        "no-seed",
+        "no-sm2",
+        "no-sm3",
+        "no-sm4",
+        "enable-ec_nistp_64_gcc_128",
+    ];
+
+    if env::var("PROFILE").unwrap().as_str() == "debug" {
+        buildargs.push("--debug");
+    }
+
+    let mut defines = "-DDEVRANDOM=\\\"/dev/urandom\\\"".to_string();
 
-    let str_defines;
+    let ar_path: std::path::PathBuf;
+    let ar_name: &str;
+
+    #[cfg(not(feature = "fips"))]
+    {
+        ar_path = openssl_path.clone();
+        ar_name = "crypto";
+    }
 
     #[cfg(feature = "fips")]
-    let (libpath, bldargs, header) = {
-        let providers_path = openssl_path
+    {
```

Comment:
nit: this is the exact use case, where the `if-cfg`/`else` would make the condition easier to read as it makes it clear on the first sight that one condition belongs to the other and one does not need to spend cognitive energy to figure out what is the difference between these two conditions. But I am ok to keep it like this if you prefer.